# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1551857034 -3600
#      Wed Mar 06 08:23:54 2019 +0100
# Node ID 156bd21bf6bdd059e21c516afc0d655ab94633c8
# Parent  ab97008be4116692cc8739a3e9167d13891b0f10
editor follows XDG_CONFIG_HOME for autocompletion files (bug ##55822)

* file-editor-tab.cc (update_lexer_settings): use the path of the
  settings file as determined at start up (honoring XDG_CONFIG_HOME)
  instead og the standard location

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -634,25 +634,19 @@ namespace octave
 
     if (_lexer_apis)
       {
         _lexer_apis->cancelPreparation ();  // stop preparing if apis exists
 
         bool update_apis = false;  // flag, whether update of apis files
 
         // get path to prepared api info
-#if defined (HAVE_QSTANDARDPATHS)
-        QString prep_apis_path
-          = QStandardPaths::writableLocation (QStandardPaths::HomeLocation)
-            + "/.config/octave/" + QString (OCTAVE_VERSION) + "/qsci/";
-#else
-        QString prep_apis_path
-          = QDesktopServices::storageLocation (QDesktopServices::HomeLocation)
-            + "/.config/octave/" + QString (OCTAVE_VERSION) + "/qsci/";
-#endif
+        QFileInfo settings_file = resource_manager::get_settings_file ();
+        QString prep_apis_path = settings_file.absolutePath () + "/"
+                                  + QString (OCTAVE_VERSION) + "/qsci/";
 
         // get settings which infos are used for octave
         bool octave_builtins = settings->value (
                                  "editor/codeCompletion_octave_builtins", true).toBool ();
         bool octave_functions = settings->value (
                                   "editor/codeCompletion_octave_functions", true).toBool ();
 
         if (_is_octave_file)
